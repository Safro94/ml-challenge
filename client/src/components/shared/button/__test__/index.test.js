import React from 'react';

import { render } from '@testing-library/react';
import { screen } from '@testing-library/dom';
import userEvent from '@testing-library/user-event';

import Button from '../';

describe('Button', () => {
	it('should render OK', () => {
		//Act
		const { container } = render(
			<Button>
				<h1>Test</h1>
			</Button>
		);

		//Assert
		expect(container).toMatchSnapshot();
	});

	it('should call onclick', () => {
		//Arrange
		const onClick = jest.fn();

		//Act
		render(
			<Button onClick={onClick}>
				<h1>Test</h1>
			</Button>
		);
		userEvent.click(screen.getByRole('button'));

		//Assert
		expect(onClick).toHaveBeenCalledTimes(1);
	});
});
