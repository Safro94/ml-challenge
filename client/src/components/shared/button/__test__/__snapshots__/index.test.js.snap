// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render OK 1`] = `
<div>
  <button
    class="button"
    type="submit"
  >
    <h1>
      Test
    </h1>
  </button>
</div>
`;
